var/spack/repos/builtin/packages/py-installer/package.py

```diff
@@ -44,6 +44,6 @@ def install(self, spec, prefix):
         python(*args)
 
     def setup_dependent_package(self, module, dependent_spec):
-        installer = dependent_spec["python"].command
+        installer = dependent_spec["python-venv"].command
         installer.add_default_arg("-m", "installer")
```

Comment:
To avoid exposing `python-venv`, I would add `copy()`  to `Executable` and do:

```suggestion
        installer = python.copy()
        installer.add_default_arg("-m", "installer")
```